ddtrace/contrib/botocore/services/bedrock.py

```diff
@@ -0,0 +1,297 @@
+        metadata = streamed_body[-1]["amazon-bedrock-invocationMetrics"]
+    elif provider == _STABILITY:
+        # TODO: figure out extraction for image-based models
+        pass
+    return {
+        "response.duration": metadata.get("invocationLatency", None),
+        "usage.prompt_tokens": metadata.get("inputTokenCount", None),
+        "usage.completion_tokens": metadata.get("outputTokenCount", None),
+    }
+
+
+def handle_bedrock_request(span: Span, integration: BedrockIntegration, params: Dict[str, Any]) -> None:
+    """Perform request param extraction and tagging."""
+    model_provider, model_name = params.get("modelId").split(".")
+    request_params = _extract_request_params(params, model_provider)
+
+    span.set_tag_str("bedrock.request.model_provider", model_provider)
+    span.set_tag_str("bedrock.request.model", model_name)
+    for k, v in request_params.items():
+        if k == "prompt" and integration.is_pc_sampled_span(span):
+            v = integration.trunc(str(v))
+        span.set_tag_str("bedrock.request.{}".format(k), str(v))
+
+
+def handle_bedrock_response(span: Span, integration: BedrockIntegration, result: Dict[str, Any]) -> Dict[str, Any]:
+    """Perform response param extraction and tagging."""
+    metadata = result["ResponseMetadata"]
+    span.set_tag_str("bedrock.response.id", str(metadata.get("RequestId", "")))
+    span.set_tag_str("bedrock.response.duration", str(metadata.get("x-amzn-bedrock-invocation-latency", "")))
+    span.set_tag_str("bedrock.usage.prompt_tokens", str(metadata.get("x-amzn-bedrock-input-token-count", "")))
+    span.set_tag_str("bedrock.usage.completion_tokens", str(metadata.get("HTTPStatusCode", "")))
+
+    # Wrap the StreamingResponse in a traced object so that we can tag response data as the user consumes it.
+    body = result["body"]
+    result["body"] = TracedBotocoreStreamingBody(body, span, integration)
+    return result
+
+
+def patched_bedrock_api_call(original_func, instance, args, kwargs, function_vars):
+    params = function_vars.get("params")
+    trace_operation = function_vars.get("trace_operation")
+    operation = function_vars.get("operation")
+    pin = function_vars.get("pin")
+    endpoint_name = function_vars.get("endpoint_name")
+    integration = function_vars.get("integration")
+    # This span will be finished separately as the user fully consumes the stream body, or on error.
+    bedrock_span = pin.tracer.start_span(
+        trace_operation,
+        service=schematize_service_name("{}.{}".format(pin.service, endpoint_name)),
+        resource=operation,
+        activate=False,
+    )
+    try:
+        handle_bedrock_request(bedrock_span, integration, params)
+        result = original_func(*args, **kwargs)
+        result = handle_bedrock_response(bedrock_span, integration, result)
+        return result
+    except Exception:
+        bedrock_span.set_exc_info(*sys.exc_info())
+        bedrock_span.finish()
+        raise
```

Comment:
```suggestion
        raise
    finally:
        bedrock_span.finish()
```

Reply:
We can't finish the span on all cases here, since the user has to consume the body before we can finish it.